Week1/homework/app.js

```diff
@@ -1,11 +1,114 @@
+    "sweet_orange_tree": { title: "My Sweet Orange Tree", author: "Vasconcelos", language: "Portuguese", publisher: "Can Publisher", "first edition": 1986 },
+    "what_men_live_by": { title: "What Men Live By?", author: "Tolstoy", language: "Russian", publisher: "Oxford", "first edition": 1878 },
+    "the_alchemist": { title: "The Alchemist", author: "Paulo Coelho", language: "Portuguese", publisher: "Metis", "first edition": 2005 }
+  };
+
+  const bookCover = {
+    "crime_and_punisment": "https://is1-ssl.mzstatic.com/image/thumb/Publication/v4/c7/5f/93/c75f93e2-9f03-82a0-3970-332d46ce3b26/source/1200x630bb.jpg",
+    "zorba": "https://images-na.ssl-images-amazon.com/images/I/81znZ1KoT6L._SY445_.jpg",
+    "siddhartha": "https://images-na.ssl-images-amazon.com/images/I/51giy3zae6L._SX329_BO1,204,203,200_.jpg",
+    "white_fang": "https://images-na.ssl-images-amazon.com/images/I/41E5RNrSKhL._SX331_BO1,204,203,200_.jpg",
+    "poor_folk": "https://images-na.ssl-images-amazon.com/images/I/51iPBDEi1SL.jpg",
+    "white_nights": "https://images-na.ssl-images-amazon.com/images/I/51Rzum6TgBL._SX331_BO1,204,203,200_.jpg",
+    "sorrows_young_werther": "https://images-na.ssl-images-amazon.com/images/I/51fBHrUsMNL.jpg",
+    "sweet_orange_tree": "https://upload.wikimedia.org/wikipedia/en/thumb/4/4d/My_Sweet_Orange_Tree_2012_poster.jpg/220px-My_Sweet_Orange_Tree_2012_poster.jpg",
+    "what_men_live_by": "https://images-na.ssl-images-amazon.com/images/I/412JkTATtBL._SX331_BO1,204,203,200_.jpg",
+    "the_alchemist": "https://images.gr-assets.com/books/1483412266l/865.jpg"
+  };
+
+
+
+  function printBookDetails(placeId, myObj, cover) {
+    let div = document.getElementById(placeId);
+
+    function size(obj) {
+      let size = 0, key;
+      for (key in obj) {
+        if (obj.hasOwnProperty(key)) size++;
+      }
+      return size;
+    };
+
+    for (let a = 0; a < size(myObj); a++) {
+      let myDiv = document.createElement("div");
+
+      div.appendChild(myDiv);
+
+      myDiv.setAttribute("class", "myDiv");
+
+      myDiv.setAttribute("id", Object.keys(myObj)[a]);
+
+      let h2 = document.createElement("h2");
+
+      myDiv.appendChild(h2);
+
+      h2.innerHTML = myObj[Object.keys(myObj)[a]].title
+
+      //h2.setAttribute('id', Object.keys(myObj)[a]);
+
+      let ul = document.createElement('ul');
+
+      myDiv.appendChild(ul);
+
+      //ul.setAttribute("id", Object.keys(myObj)[a]);
+
+      let img = document.createElement("img");
+
+      myDiv.appendChild(img);
+
+      img.setAttribute("src", Object.values(cover)[a]);
+
+      img.setAttribute("alt", myObj[Object.keys(myObj)[a]].title);
+
+      for (let k = 1; k < 5; k++) {
+        let li = document.createElement('li');
+        ul.appendChild(li);
+        li.innerHTML = (Object.keys(myObj[Object.keys(myObj)[a]])[k] + ": " + Object.values(myObj[Object.keys(myObj)[a]])[k]);
```

Comment:
I think here is the best example I can introduce for you about the important of a meaningful names.
In your code in best cases with the names you have we can improve it by order like so:
```javascript
li.innerHTML = (
  Object.keys(
      myObj[
        Object.keys(myObj)[a] // Magic 👻
      ]
  )[k] + ": " + Object.values(
      myObj[
        Object.keys(myObj)[a]
      ]
  )[k]
);
```
And it still Not a pattern and obviously not something by the first look as a developer legible.
so what you can do is:
 - to define the `Object.keys` with a meaningful name like "booksNames" *(you may have a better name)* before the first for loop even *(you used it multiple times)*
 - then use it everywhere you are using it now

even if this one step didn't fix that much but you will still have cleaner code and more readable.